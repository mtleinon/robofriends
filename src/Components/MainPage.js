import React, {Component} from 'react';
import { CardList } from "../Components/CardList";
import { SearchBox } from '../Components/SearchBox';
import { Scroll } from '../Components/Scroll';
import { ErrorBoundary } from "../Containers/ErrorBoundary";
import './MainPage.css';

import {Header} from "../Containers/Header";

class MainPage extends Component {

    componentDidMount() {
        this.props.onRequestRobots();
    }

    filteredRobots = () =>
        this.props.robots.filter(robot =>
                robot.name.toLowerCase().includes(
                        this.props.searchField.toLowerCase()));

    render() {
        const { onSearchChange, robots, isPending } = this.props;

        if (isPending || robots.length === 0) {
            return(<div className='tc'>
                <Header/>
                <h2>Loading robots ...</h2>
            </div>);
        } else {
            return <div className='tc'>
                <Header/>
                <SearchBox searchChange={onSearchChange}/>
                <Scroll>
                    <ErrorBoundary>
                        <CardList robots={this.filteredRobots()}/>
                    </ErrorBoundary>
                </Scroll>
            </div>;
        }
    }
}

export default MainPage;