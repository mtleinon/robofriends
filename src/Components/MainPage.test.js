import React from 'react';
import { shallow } from 'enzyme';
import MainPage  from './MainPage';


let wrapper;
let mockProps;
beforeEach (() => {

    mockProps = {
        onRequestRobots: jest.fn,
        robots: [{
            id: 1,
            name: 'John',
            email: 'john@gmail.com'
        },
        {
            id: 2,
            name: 'Jane',
            email: 'jane@gmail.com'
        },
        {
            id: 3,
            name: 'Lisa',
            email: 'lisa@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    };
    wrapper = shallow(<MainPage { ...mockProps}/>);

});

it('MainPage is rendered', () => {
    expect(wrapper).toMatchSnapshot();
});

it('MainPage is rendered when isPending is true', () => {
    const wrapper2 = shallow(<MainPage { ...mockProps} isPending={true}/>);
    expect(wrapper2).toMatchSnapshot();
});

it('filterRobots works, a robot found', () => {
    expect(wrapper.instance().filteredRobots()).toEqual([{
        id: 1,
        name: 'John',
        email: 'john@gmail.com'
    }]);
});

it('filterRobots works, a robot not found', () => {
    const wrapper2 = shallow(<MainPage { ...mockProps} searchField={'aa'}/>);

    expect(wrapper2.instance().filteredRobots()).toEqual([]);
});
it('filterRobots works, two robots are found', () => {
    const wrapper2 = shallow(<MainPage { ...mockProps} searchField={'J'}/>);

    expect(wrapper2.instance().filteredRobots()).toEqual([{
        id: 1,
        name: 'John',
        email: 'john@gmail.com'
    },
    {
        id: 2,
        name: 'Jane',
        email: 'jane@gmail.com'
    }]);
});
