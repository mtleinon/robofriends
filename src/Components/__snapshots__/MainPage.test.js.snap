// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage is rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={[Function]}
    robots={
      Array [
        Object {
          "email": "john@gmail.com",
          "id": 1,
          "name": "John",
        },
        Object {
          "email": "jane@gmail.com",
          "id": 2,
          "name": "Jane",
        },
        Object {
          "email": "lisa@gmail.com",
          "id": 3,
          "name": "Lisa",
        },
      ]
    }
    searchField="john"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchBox />,
        <Scroll>
          <ErrorBoundary>
            <CardList
              robots={
                Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 1,
                    "name": "John",
                  },
                ]
              }
            />
          </ErrorBoundary>
        </Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundary>
            <CardList
              robots={
                Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 1,
                    "name": "John",
                  },
                ]
              }
            />
          </ErrorBoundary>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={
                Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 1,
                    "name": "John",
                  },
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [
                Object {
                  "email": "john@gmail.com",
                  "id": 1,
                  "name": "John",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchBox />,
          <Scroll>
            <ErrorBoundary>
              <CardList
                robots={
                  Array [
                    Object {
                      "email": "john@gmail.com",
                      "id": 1,
                      "name": "John",
                    },
                  ]
                }
              />
            </ErrorBoundary>
          </Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundary>
              <CardList
                robots={
                  Array [
                    Object {
                      "email": "john@gmail.com",
                      "id": 1,
                      "name": "John",
                    },
                  ]
                }
              />
            </ErrorBoundary>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={
                  Array [
                    Object {
                      "email": "john@gmail.com",
                      "id": 1,
                      "name": "John",
                    },
                  ]
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [
                  Object {
                    "email": "john@gmail.com",
                    "id": 1,
                    "name": "John",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`MainPage is rendered when isPending is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={true}
    onRequestRobots={[Function]}
    robots={
      Array [
        Object {
          "email": "john@gmail.com",
          "id": 1,
          "name": "John",
        },
        Object {
          "email": "jane@gmail.com",
          "id": 2,
          "name": "Jane",
        },
        Object {
          "email": "lisa@gmail.com",
          "id": 3,
          "name": "Lisa",
        },
      ]
    }
    searchField="john"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <h2>
          Loading robots ...
        </h2>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading robots ...",
        },
        "ref": null,
        "rendered": "Loading robots ...",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <h2>
            Loading robots ...
          </h2>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading robots ...",
          },
          "ref": null,
          "rendered": "Loading robots ...",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
